stixpy/timeseries/anc.py

```diff
@@ -0,0 +1,205 @@
+    """
+
+    _source = "stix"
+
+    def plot(self, axes=None, columns=None, **plot_args):
+        r"""
+        Show a plot of the data.
+
+        Parameters
+        ----------
+        axes : `~matplotlib.axes.Axes`, optional
+            If provided the image will be plotted on the given axes.
+            Defaults to `None`, so the current axes will be used.
+        columns : `list`, optional
+            Columns to plot, defaults to 'counts'.
+        **plot_args : `dict`, optional
+            Additional plot keyword arguments that are handed to
+            :meth:`pandas.DataFrame.plot`.
+
+        Returns
+        -------
+        axes : `~matplotlib.axes.Axes`
+            The plot axes.
+        """
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        from sunpy.net import attrs as a
+
+        if columns is None:
+            columns = ["y_srf", "z_srf"]
+            axes, columns = self._setup_axes_columns(axes, columns)
+
+            sas_ok_values = self.data["sas_ok"]
+            has_false = np.any(~sas_ok_values)
+
+            if has_false:
+                # copy data to destroy reference and keep original data
+                y_srf_cp_t = self.to_dataframe()["y_srf"].copy().to_numpy()
+                z_srf_cp_t = self.to_dataframe()["z_srf"].copy().to_numpy()
+                # overwrite False data with nan such that there is a break in the plot when results are invalid
+                y_srf_cp_t[self.data["sas_ok"] == False] = np.nan  # noqa:  E712
+                z_srf_cp_t[self.data["sas_ok"] == False] = np.nan  # noqa:  E712
+
+                # copy data to destroy reference and keep original data
+                y_srf_cp_f = self.to_dataframe()["y_srf"].copy().to_numpy()
+                z_srf_cp_f = self.to_dataframe()["z_srf"].copy().to_numpy()
+                # overwrite True data with nan such that there is a break in the plot when results are valid
+                y_srf_cp_f[self.data["sas_ok"] == True] = np.nan  # noqa:  E712
+                z_srf_cp_f[self.data["sas_ok"] == True] = np.nan  # noqa:  E712
+
+                # plot graphs where sas_ok is True with a solid line
+                a = axes.plot(self._data.index, y_srf_cp_t, linestyle="solid", label="sun y = valid", **plot_args)
+                b = axes.plot(self._data.index, z_srf_cp_t, linestyle="solid", label="sun z = valid", **plot_args)
+
+                a = a[0].get_color()
+                b = b[0].get_color()
+
+                # plot graphs where sas_ok is False with a dashed line
+                axes.plot(
+                    self._data.index, y_srf_cp_f, linestyle="dashed", color=a, label="sun y = invalid", **plot_args
+                )
+                axes.plot(
+                    self._data.index, z_srf_cp_f, linestyle="dashed", color=b, label="sun z = invalid", **plot_args
+                )
+
+            else:
+                axes = self._data[columns].plot(ax=axes, **plot_args)
```

Comment:
This seems very complex I think something like this should work

```
sas_ok = self.data["sas_ok"].values
a = axes.plot(self._data.index[sas_ok], self._data.["y_srf"][sas_ok],
              linestyle="solid", label="sun y = valid", **plot_args)
...
axes.plot(self._data.index[~sas_ok], self._data.["y_srf"][~sas_ok],
          linestyle="dashed", color=a, label="sun y = invalid", **plot_args)
```

Reply:
i remember that  we tried a similar compact approach but with that the line did nit had a break between valid and invalid time period it was connected by a straight line.  Is that not the case with your approach?

Reply:
ok this looks like intended. if this is with the compact code - then absolutely go for it 
<img width="640" height="537" alt="grafik" src="https://github.com/user-attachments/assets/c1802fbf-cc29-4816-8aaf-a0c7607d603e" />
i was afraid that with this approach the gab is closed by a connected line

Reply:
Ah yea ok - no still uses your nan approach just accomplished in a different way. I might make some small changes to the plot legend then it should be good to go.